delocate/tools.py

```diff
@@ -246,18 +369,30 @@ def get_install_names(filename):
     install_names : tuple
         tuple of install names for library `filename`
     """
-    lines = _cmd_out_err(["otool", "-L", filename])
-    if not _line0_says_object(lines[0], filename):
+    otool = subprocess.run(
+        ["otool", "-L", filename],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    if not _line0_says_object(otool.stdout or otool.stderr, filename):
         return ()
-    names = tuple(parse_install_name(line)[0] for line in lines[1:])
-    install_id = get_install_id(filename)
-    if install_id is not None:
-        assert names[0] == install_id
-        return names[1:]
-    return names
-
-
-def get_install_id(filename):
+    install_ids = _get_install_ids(filename)
+    all_names = []
+    for arch, libraries in _parse_otool_install_names(otool.stdout).items():
+        names_for_arch = [name for name, _, _ in libraries]
+        if arch in install_ids:
+            # Remove redundant install id from the install names.
+            assert names_for_arch[0] == install_ids[arch]
```

Comment:
Assert to error (can be optimized away)?

Reply:
This was the original behavior.  I don't see any way this could be False, but I'd like to keep the assert just in case.

Reply:
I've refactored the code.  It's less complicated now.